pkg/errors/user_errors.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"errors"
+)
+
+type UserMessenger interface {
```

Comment:
If the idea is that the caller should use the exported `UserErrorWithMessage()` to fetch the displayable message, is this interface necessary? The Go stdlib exposes many [errors by value](https://pkg.go.dev/io#pkg-variables).

Reply:
Yeah, I agree with the points. But if the interface is really required, I'd put a description on top for the future implementations (if any). Also, we should probably add `Unwrap` to the interface to guarantee that the implementing errors implement it as well.